repos/system_upgrade/el7toel8/actors/nischeck/libraries/nischeck.py

```diff
@@ -0,0 +1,152 @@
+import re
+import ipaddress
+
+from leapp import reporting
+
+
+def scan():
+    """
+    Check configuration of /etc/yp.conf and /etc/nsswitch.conf. Address of NIS server cannot be
+    specified by host name if NIS is used to resolve host names.
+    """
+
+    hostnames = hostnames_in_yp_conf()
+    data, _ = scan_nsswitch()
+
+    if "hosts" in data:
+        if "nis" in data["hosts"] and hostnames:
+            report_error(hostnames)
+
+
+def report_error(hostnames):
+    reporting.create_report([
+        reporting.Title('Unsupported NIS configuration found'),
+        reporting.Summary("NIS may be used for domain name resolution only if NIS "
+                          "server is specified by IP. NIS servers specified by "
+                          "host name: {}".format(", ".join(hostnames))),
+        reporting.Severity(reporting.Severity.MEDIUM)
+    ])
+
+
+def hostnames_in_yp_conf():
+    """
+    Scan /etc/yp.conf, return if file contains domain name
+    """
+
+    path = "/etc/yp.conf"
+    try:
+        with open(path) as f:
+            lines = f.readlines()
+    except IOError:
+        return False
```

Comment:
Inconsistent return type - you have False here but generally the function return a list of not_ips. Please return same type from all branches.